packages/structures/src/Structure.ts

```diff
@@ -0,0 +1,133 @@
+
+	/**
+	 * The template used for removing data from the raw data stored for each Structure.
+	 *
+	 * @remarks This template should be overriden in all subclasses to provide more accurate type information.
+	 * The template in the base {@link Structure} class will have no effect on most subclasses for this reason.
+	 */
+	protected static DataTemplate: Record<string, unknown> = {};
+
+	/**
+	 * @returns A cloned version of the data template, ready to create a new data object.
+	 */
+	private _getDataTemplate() {
+		return Object.create((this.constructor as typeof Structure).DataTemplate);
+	}
+
+	/**
+	 * The raw data from the API for this struture
+	 *
+	 * @internal
+	 */
+	protected [kData]: Readonly<ReplaceOmittedWithUnknown<Omitted, DataType>>;
+
+	/**
+	 * Creates a new structure to represent API data
+	 *
+	 * @param data - the data from the API that this structure will represent
+	 * @param extraOptions - any additional options needed to appropriately construct a structure from the data
+	 * @remarks To be made public in subclasses
+	 * @internal
+	 */
+	protected constructor(data: Readonly<Partial<DataType>>, ..._rest: unknown[]) {
+		this[kData] = Object.assign(this._getDataTemplate(), data);
+		this[kMixinConstruct]?.(data);
+	}
+
+	/**
+	 * Patches the raw data of this object in place
+	 *
+	 * @param data - the updated data from the API to patch with
+	 * @param extraOptions - any additional options needed to appropriately patch the data
+	 * @remarks To be made public in subclasses
+	 * @returns this
+	 * @internal
+	 */
+	protected _patch(data: Readonly<Partial<DataType>>): this {
+		this[kData] = Object.assign(this._getDataTemplate(), this[kData], data);
+		this._optimizeData(data);
+		return this;
+	}
+
+	/**
+	 * Creates a clone of this structure
+	 *
+	 * @returns a clone of this
+	 * @internal
+	 */
+	protected _clone(): typeof this {
+		// @ts-expect-error constructor is of abstract class is unknown
+		return new this.constructor(this.toJSON());
+	}
+
+	/**
+	 * Function called to ensure stored raw data is in optimized formats, used in tandem with a data template
+	 *
+	 * @example created_timestamp is an ISO string, this can be stored in optimized form as a number
+	 * @param _data - the raw data received from the API to optimize
+	 * @remarks Implementation to be done in subclasses and mixins where needed.
+	 * For typescript users, mixins must use the closest ancestors access modifier.
+	 * @remarks Automatically called in {@link Structure#_patch} but must be called manually in the constructor
+	 * of any class implementing this method.
+	 * @remarks Additionally, when implementing, ensure to call `super._optimizeData` if any class in the super chain aside
+	 * from Structure contains an implementation.
+	 * Note: mixins do not need to call super ever as the process of mixing walks the prototype chain.
+	 * @virtual
+	 * @internal
+	 */
+	protected _optimizeData(_data: Partial<DataType>) {}
```

Comment:
Why not `abstract`? So classes that don't have any work to do here don't need to implement? Not sure I love that, we could go this route instead if we want to avoid empty implementations everywhere:
```suggestion
	protected _optimizeData?(_data: Partial<DataType>): void;
```
and calls do `this._optimizeData.?();` (`kMixinConstruct` also follows this pattern)

(notice the lack of an `abstract` keyword, that actually requires explicitly specifying it in the subclass even if you don't want to implement it)

Reply:
This method gets defined/overridden in the Mixin process. Making it optional instead of an empty function body would mean that every implementation of a Structure would have to do the optional chaining of that method call in their constructor. `kMixinConstruct` works completely different, as it's not a method that gets overridden but instead all of them get combined into a single function that gets called in the `Structure` base class here if present. Implementers don't need to worry about that optionality at all.